lib/ain-grpc/src/transaction_request.rs

```diff
@@ -0,0 +1,99 @@
+use crate::bytes::Bytes;
+use ethereum::{
+    AccessListItem, EIP1559TransactionMessage, EIP2930TransactionMessage, LegacyTransactionMessage,
+};
+use ethereum_types::{H160, U256};
+use serde::{Deserialize, Serialize};
+
+pub enum TransactionMessage {
+    Legacy(LegacyTransactionMessage),
+    EIP2930(EIP2930TransactionMessage),
+    EIP1559(EIP1559TransactionMessage),
+}
+
+/// Transaction request coming from RPC
+#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "camelCase")]
+pub struct TransactionRequest {
+    /// Sender
+    pub from: Option<H160>,
+    /// Recipient
+    pub to: Option<H160>,
+    /// Gas Price, legacy.
+    #[serde(default)]
+    pub gas_price: Option<U256>,
+    /// Max BaseFeePerGas the user is willing to pay.
+    #[serde(default)]
+    pub max_fee_per_gas: Option<U256>,
+    /// The miner's tip.
+    #[serde(default)]
+    pub max_priority_fee_per_gas: Option<U256>,
+    /// Gas
+    pub gas: Option<U256>,
+    /// Value of transaction in wei
+    pub value: Option<U256>,
+    /// Additional data sent with transaction
+    pub data: Option<Bytes>,
+    /// Transaction's nonce
+    pub nonce: Option<U256>,
+    /// Pre-pay to warm storage access.
+    #[serde(default)]
+    pub access_list: Option<Vec<AccessListItem>>,
+    /// EIP-2718 type
+    #[serde(rename = "type")]
+    pub transaction_type: Option<U256>,
+}
+
+impl From<TransactionRequest> for Option<TransactionMessage> {
```

Comment:
Would be better to implement TryFrom here and have it return an error instead of None on last match arm

Reply:
noted that.. its a great practice.. but we ady have error handling in another way round
https://github.com/DeFiCh/ain/blob/fde4c9d71b41502533d4569e35affb2de1534cad/lib/ain-grpc/src/rpc.rs#L543-L547